e2e/cypress/integration/enterprise/system_console/channel_moderation_spec.js

```diff
@@ -35,24 +59,876 @@ const enableAllChannelModeratedPermissions = () => {
+            cy.findByTestId('admin-channel_settings-channel_moderation-channelMentions-disabledGuestsDueToCreatePosts').
+                should('have.text', 'Guests can not use channel mentions without the ability to create posts.');
+            cy.findByTestId(checkboxesTitleToIdMap.CHANNEL_MENTIONS_GUESTS).should('be.disabled');
+
+            // # check the create posts option for guests and uncheck for memebers
+            enableChannelModeratedPermission(checkboxesTitleToIdMap.CREATE_POSTS_GUESTS);
+            disableChannelModeratedPermission(checkboxesTitleToIdMap.CREATE_POSTS_MEMBERS);
+
+            // * Option to allow Channel Mentions for Members should also be disabled when Create Post option is disabled.
+            // * A message Members can not use channel mentions without the ability to create posts should be displayed.
+            cy.findByTestId('admin-channel_settings-channel_moderation-channelMentions-disabledMemberDueToCreatePosts').
+                should('have.text', 'Members can not use channel mentions without the ability to create posts.');
+            cy.findByTestId(checkboxesTitleToIdMap.CHANNEL_MENTIONS_MEMBERS).should('be.disabled');
+
+            // # Uncheck the create posts option for guests
+            disableChannelModeratedPermission(checkboxesTitleToIdMap.CREATE_POSTS_GUESTS);
+
+            // * Ensure that channel mentions for members and guests is disabled
+            // * Ensure message Guests & Members can not use channel mentions without the ability to create posts
+            cy.findByTestId('admin-channel_settings-channel_moderation-channelMentions-disabledBothDueToCreatePosts').
+                should('have.text', 'Guests and members can not use channel mentions without the ability to create posts.');
+            cy.findByTestId(checkboxesTitleToIdMap.CHANNEL_MENTIONS_GUESTS).should('be.disabled');
+            cy.findByTestId(checkboxesTitleToIdMap.CHANNEL_MENTIONS_MEMBERS).should('be.disabled');
+        });
+
+        it('Message when user without channel mention permission uses special channel mentions', () => {
+            visitAutemChannelConfigPage();
+            disableChannelModeratedPermission(checkboxesTitleToIdMap.CHANNEL_MENTIONS_MEMBERS);
+            saveConfig();
+
+            visitAutemChannel('user-1');
+
+            cy.findByTestId('post_textbox').clear().type('@');
+
+            // * Ensure that @here, @all, and @channel do not show up in the autocomplete list
+            cy.findAllByTestId('mentionSuggestion_here').should('not.exist');
+            cy.findAllByTestId('mentionSuggestion_all').should('not.exist');
+            cy.findAllByTestId('mentionSuggestion_channel').should('not.exist');
+
+            // * When you type @all, @enter, and @channel make sure that a system message shows up notifying you nothing happened.
+            postChannelMentionsAndVerifySystemMessageExist();
```

Comment:
A step is missing before this expected result. It should type @all and submit. Only then the system message would appear. 
`cy.findByTestId('post_textbox').clear().type('@all{enter}');`

Reply:
However `postChannelMentionsAndVerifySystemMessageExist` should already be doing that. 
```
// # Checks to see if we got a system message warning after using @all/@here/@channel
const postChannelMentionsAndVerifySystemMessageExist = () => {
    // # Type @all and post it to the channel
    cy.findByTestId('post_textbox').clear().type('@all{enter}');

    // # Get last post message text
    cy.getLastPostId().then((postId) => {
        // * Assert that the last message posted is the system message informing us we are not allowed to use channel mentions
        cy.get(`#postMessageText_${postId}`).should('include.text', 'Channel notifications are disabled in aut-8. The @all did not trigger any notifications.');
    });
...
```
We can see that in that function, the first thing we do is post the `@all`. Was this not happening?